### ISSUE TITLE
Create new gradle task that runs all the detekt tasks

### ISSUE THREAD

**Issue by mkrussel77** [detekt/detekt#3838]
Your feature request may already be reported!
Please search in the [issues](https://github.com/detekt/detekt/issues) before creating a new one.

## Expected Behavior

Would be nice for there to be a single task that runs all the other Detekt tasks.

So with an Android project it would run

- detekt
- detektMain
- detektTest

For Multiplatform it would run
- detekt
- detektAndroidDebug
- detektAndroidDebugAndroidTest
- detektAndroidDebugUnitTest
- detektAndroidRelease
- detektAndroidReleaseAndroidTest
- detektAndroidReleaseUnitTest
- detektIosArm64Main
- detektIosArm64Test
- detektIosX64Main
- detektIosXcd64Test
- detektJvmMain
- detektJvmTest
...

## Current Behavior

There is no task

## Context

Goal would be to simplify the number of tasks to enter to get the type resolution runs to run on all the source code.

Right now it is easy for Android projects, but Multiplatform projections are complicated.

**Comment by BraisGabin** [comment]
Agree. I propose to use `detektCheck` for that. I think that `detektMain` and `detektTest` should be internal tasks, not intended for the use of the users. And, probably, it should output an unified report of that module instead of two. 

But I'm not sure if we are at the point to provide it without adding breaking changes in the future. 

I recommend you to create a custom task that depend on all those rules so you don't need to write all of those each time. 

Side note: If you run `detektMain` and `detektTest` you don't need to run `detekt`.

**Comment by cortinico** [comment]
I believe we could add a `detektAll` task that depends on all the `detekt*` tasks, but **not** on the `detekt` task (as it could produce conflicting results having Type Resolution enabled/disabled.

**Comment by mkrussel77** [comment]
After creating this ticket, I found out that everything is added to the `check` task.  So that is one option.

I'm currently having problems creating my own task due to #3840

**Comment by 3flex** [comment]
> I found out that everything is added to the check task

This is the Gradle convention for code quality tasks - individual tasks are added as dependencies of the `check` lifecycle task.

The problem with adding new tasks with "detekt" prefix that depend on other detekt tasks is that if there's a source set named "check" for example and this plugin had (again, example) a task called "detektCheck" that just ran other detekt tasks then there'd be a collision on task names.

So I don't support adding a new task that just depends on other detekt tasks. But an example of how to configure this and attach tasks to `check`, or create a custom task for that purpose, can be added to the documentation.

**Comment by BraisGabin** [comment]
So, which is the follow up for this issue? What should we do to close it?

**Comment by cortinico** [comment]
> I found out that everything is added to the `check` task

It turns out it's not, that was a "bug" of the KMP implementation, as the Android tasks are not added to `check`.

I'm proposing a solution to this issue here: https://github.com/detekt/detekt/issues/3927#issuecomment-894863396

**Comment by eygraber** [comment]
Is there any plan to get this into an alpha soon? I rolled my own in v1, but with the changes to how Detekt tasks are registered for KMP it got much harder to do from the outside in v2.

**Comment by BraisGabin** [comment]
Once #7018 is implemented this would be fixed, right?

**Comment by cortinico** [comment]
Closing as duplicate of #7018
